### PR TITLE
ROC-5059: Mediation phone number ccd

### PR DESCRIPTION
**Pull request by KiranVarma02**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5059

### Change description ###
mediation phone number ccd changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[* ] No
```

### CODE REVIEW

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ResponseMapperTest.java [hunk 1]
```diff
@@ -85,6 +85,19 @@ public void shouldMapFullDefenceResponseFromCCD() {
...
+    public void shouldMapFullDefenceResponseWithFreeMediationFromCCD() {
+        //given
+        CCDDefendant ccdDefendant = SampleCCDDefendant.withFullDefenceResponseAndFreeMediation().build();
+        Claim.ClaimBuilder builder = Claim.builder();
+
+        //when
+        mapper.from(builder, ccdDefendant);
+
+        //then
+        assertThat(builder.build().getResponse().orElse(null)).isEqualTo(ccdDefendant);
```
Comment (dharmendrak) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ResponseMapperTest.java]: ResponseAssert need to be updated for `MediationPhoneNumber`

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/defendant/CCDDefendant.java [hunk 2]
```diff
@@ -74,6 +74,7 @@
     private CCDDefenceType responseDefenceType;
     private String responseDefence;
     private CCDYesNoOption responseFreeMediationOption;
+    private String responseMediationPhoneNumber;
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/defendant/CCDDefendant.java]: I can if you want but most other properties are starting with response, don't know why....i am just following that..

**Comment by sabahirfan** [comment]
This PR will require change and release in definition file as well.

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ResponseMapperTest.java (hunk 1 above)
Comment (KiranVarma02) [reply ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ResponseMapperTest.java]: done

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/claimantresponse/CCDResponseRejection.java [hunk 3]
```diff
@@ -13,19 +13,25 @@
...
+    private String mediationContactPerson;
     private String reason;
 
     @Builder
     public CCDResponseRejection(
         BigDecimal amountPaid,
         LocalDateTime submittedOn,
         CCDYesNoOption freeMediationOption,
+        String mediationPhoneNumber,
+        String mediationContactPerson,
```
Comment (karthikkannankainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/claimantresponse/CCDResponseRejection.java]: mediationContactPerson not in CCD Definition. Please check.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/claimantresponse/CCDResponseRejection.java [hunk 4]
```diff
@@ -13,19 +13,25 @@
 public class CCDResponseRejection extends CCDClaimantResponse {
 
     private CCDYesNoOption freeMediationOption;
+    private String mediationPhoneNumber;
```
Comment (karthikkannankainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/claimantresponse/CCDResponseRejection.java]: why do we need these two fields in claimantresponse and under ccd defendant?

Comment (KiranVarma02) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/claimantresponse/CCDResponseRejection.java]: not sure if i understand, my understanding is, defendant and claimant has different mediationPhoneNumber's, they both are different. This is how its modelled in front end and in claim store. What do you suggest we do? ping me when you are free, we will chat about this...

Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/claimantresponse/CCDResponseRejection.java]: @KiranVarma02 cool, All these confusion starts in CCD world where we have everything under CCDDefendant. The ClaimantResponse is also under CCDDefendant, so was wondering if its same. My bad.

**Pull request closed without merging**